providers/signalk-ws.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Ilker Temir <ilker@ilkertemir.com> based on code
+ * by Teppo Kurki <teppo.kurki@iki.fi>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var Transform = require('stream').Transform;
+
+var SignalK = require('signalk-client');
+
+var debug = require('debug')('signalk-server:providers:signalk-ws');
+
+var WebSocket = require('ws');
+var _object = require('lodash/object');
+
+function SignalKWs(options) {
+  Transform.call(this, {
+    objectMode: true
+  });
+  this.selfHost = options.app.config.getExternalHostname() + ".";
+  this.selfPort = options.app.config.getExternalPort();
+  this.remoteServers = {};
+  this.remoteServers[this.selfHost + ":" + this.selfPort] = {};
+  this.signalkClient = new SignalK.Client();
+  var url = "ws://" + options.host + ":" + options.port + options.path;
```

Comment:
Sorry, I think it explains it, I went back and forth on this one and looks like I ended up overlooking it. 

This will require me to change my server though as it doesn't currently support the GET. I am not sure I want to do that but then it makes this dependent on my config, probably not the best thing. I may just close this pull request. 

Need to go now but  will circle back on it.